### PR TITLE
Switch to python 3

### LINKED ISSUE #533: Migrate to python3
Tracking issues and changes encountered while migrating to python3.
This will not track the automatic changes made through the `futurize` project, only the subsequent manual changes.

### PR DESCRIPTION
**Pull request by shankari**
Long set of changes to switch to python 3
- multiple runs of futurize
- lots of really annoying bson issues because of the int change
- some warning fixes

This fixes #533, which has a lot more details of the errors encountered.

Tests pass on python3.
Tests almost pass on python2 - the errors are more related to the tests than the actual code.

Nevertheless, it is easy enough with anaconda to just use python3 and we should do that.
Added support for creating a conda environment for emission, made it be python3 and updated the README accordingly.

### CODE REVIEW

**Comment by shankari** [comment]
python2.7 failures attached.
[python27_failures.log.gz](https://github.com/e-mission/e-mission-server/files/1540442/python27_failures.log.gz)

**Comment by shaneknapp** [comment]
test this please

**Comment by shaneknapp** [comment]
test this please

**Comment by shaneknapp** [comment]
test this please

File: setup/teardown_tests.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+echo "Removing environment from "${CONDA_TEMP_PREFIX}
+source deactivate ${CONDA_TEMP_PREFIX}
+conda env remove --prefix ${CONDA_TEMP_PREFIX}
```
Comment (shaneknapp) [inline setup/teardown_tests.sh]: add "--yes" after "remove"

File: setup/teardown_tests.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+unset CONDA_TEMP_PREFIX
+rm conf/net/int_service/giles_conf.json
+rm conf/net/ext_service/habitica_conf.json
+rm conf/net/ext_service/push_conf.json
```
Comment (shaneknapp) [inline setup/teardown_tests.sh]: "rm -f" is your friend here!

File: setup/teardown_tests.sh [hunk 3]
```diff
@@ -2,8 +2,8 @@
...
 echo "Removing environment from "${CONDA_TEMP_PREFIX}
 source deactivate ${CONDA_TEMP_PREFIX}
-conda env remove --prefix ${CONDA_TEMP_PREFIX}
+conda env remove --yes --prefix ${CONDA_TEMP_PREFIX} 
 unset CONDA_TEMP_PREFIX
 rm conf/net/int_service/giles_conf.json
-rm conf/net/ext_service/habitica_conf.json
-rm conf/net/ext_service/push_conf.json
+rm conf/net/ext_service/habitica.json
+rm conf/net/ext_service/push.json
```
Comment (shaneknapp) [inline setup/teardown_tests.sh]: also, why do you explicitly remove these?  the build always performs a 'git clean -fdx' before re-cloning the repo.

Comment (shankari) [reply setup/teardown_tests.sh]: no strong reason - just for symmetry with the setup, where I create those files.

**Pull request merged**